nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -996,6 +999,15 @@ public static String toString(final Object value, final Supplier<DateFormat> for
             }
         }
 
+        if (value instanceof MapRecord) {
+            Map<String, Object> map = ((MapRecord)value).toMap();
+            try {
+                return MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(map);
+            } catch (Exception ex) {
+                throw new IllegalTypeConversionException("Cannot convert value " + value + " of type " + value.getClass() + " to a valid String", ex);
+            }
+        }
```

Comment:
This approach forces the implementation into a JSON-specific strategy, which is not necessarily applicable to other record-oriented data formats.